apps/www/components/Payment/PaymentRequest/PaymentRequestEventHelper.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  PaymentMethod,
+  PaymentRequestPaymentMethodEvent,
+} from '@stripe/stripe-js'
+
+type Address = {
+  name: string
+  line1: string
+  line2?: string
+  postalCode: string
+  city: string
+  country: string
+}
+
+type PayerInformation = {
+  email: string
+  firstName: string
+  lastName: string
+  shippingAddress?: Address
+}
+
+export function getPayerInformationFromEvent(
+  event: PaymentRequestPaymentMethodEvent,
+): PayerInformation {
+  const [firstName, lastName] = event.payerName.split(' ').map((s) => s.trim())
+
+  return {
+    email: event.payerEmail,
+    firstName,
+    lastName,
```

Comment:
Let's save everything. And I'd gamble that multiple first names are more common than space separated last names like `von Wunschheim`:

```suggestion
  const [...firstNames, lastName] = event.payerName.split(' ')

  return {
    email: event.payerEmail,
    firstName: firstNames.join(' ').trim(),
    lastName: lastName.trim(),
```